plutus-benchmark/marlowe/README.md

```diff
@@ -17,31 +23,27 @@ It would be informative for both the Plutus and Marlowe teams to investigate in
 
 (1) Benchmarking: compare the budget before and after optimizations that the Plutus team implemented. It could be helpful to do the benchmarking *as* we implement the optimization even.
 
-The benchmarking portion of the code lives in `marlowe/bench`(TODO), which depends on this package.
+The benchmarking portion of the code lives in `marlowe/bench`, which depends on this package.
 
 (2) Profiling: look at each script in more detail, what functions are taking up the most budget? How can they be optimized?
 
 See [CONTRIBUTING.md](https://github.com/input-output-hk/plutus/blob/master/CONTRIBUTING.adoc#how-to-build-the-code-with-profiling) for profiling instructions.
 
 Of the most common Marlowe transactions, input application transactions are the most relevant, as they are complex and can go over the execution limits at times. So there is a priority on examining those contracts.
 
-## Managing versions
+## Benchmarking with `marlowe`
 
-### Versioning of this package 
+To benchmark the Marlowe scripts, run `cabal bench marlowe`. The default time to run each benchmark is 5 seconds. One can change this with `-L` or `--timeout` when running it locally.
```

Comment:
For each benchmark Criterion runs it as many times as it can within the `-L` argument and then analyses the collected results to get the mean time and the standard deviation and so on.  So if your benchmark takes about 20ms then with the default "timeout" of 5 seconds the statistics will be based on about 250 executions.  If you've got a long-running benchmark then you might want to increase the timeout to get more accurate statistics.

Reply:
do you think 5 secs is ok for this? I'm setting it low because I'm worried about the longitudinal benchmarking workflow.

Reply:
Comment added: https://github.com/input-output-hk/plutus/pull/5415/commits/f567cfc8d1ba13e60dc149bc047f0f120be6a914

Reply:
> do you think 5 secs is ok for this? I'm setting it low because I'm worried about the longitudinal benchmarking workflow.

Aha!  The benchmarking here uses a modified Criterion configuration that records the number of iterations of each benchmark in `plutus-benchmark/report.html`, so if you look at that you can see how many times each benchmark got run.  On my laptop most of the Marlowe examples are run hundreds or thousands of times, but there are a few that are run less than 100 times, the slowest ones being run 10 and 15 times.  Those particular ones may give slightly unreliable figures, but if we turn the timeout up globally the benchmarks would take forever.  It might be possible to increase the timeout for specific benchmarks, but that would take some work.  I think we should just wait and see if there's a problem before trying to fix it.